File: composeApp/src/androidMain/kotlin/com/dnfapps/arrmatey/ui/components/HistoryItemView.kt
package com.dnfapps.arrmatey.ui.components

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.dnfapps.arrmatey.arr.api.model.HistoryItem
import com.dnfapps.arrmatey.arr.api.model.ListenarrHistoryItem
import com.dnfapps.arrmatey.compose.utils.breakable
import com.dnfapps.arrmatey.compose.utils.singleLanguageLabel
import com.dnfapps.arrmatey.entensions.bullet
import com.dnfapps.arrmatey.utils.format
import com.dnfapps.arrmatey.utils.mokoString
import kotlin.time.ExperimentalTime

@OptIn(ExperimentalTime::class)
@Composable
fun HistoryItemView(
    item: HistoryItem,
    modifier: Modifier = Modifier,
) {
    ContainerCard(
        modifier = modifier.fillMaxWidth(),
    ) {
        Row(
            horizontalArrangement = Arrangement.spacedBy(12.dp),
        ) {
            Text(
                text = mokoString(item.eventType.resource),
                color = MaterialTheme.colorScheme.primary,
                style = MaterialTheme.typography.labelMedium,
            )
            Text(
                text = item.date.format("MMM d, yyyy"),
                style = MaterialTheme.typography.labelSmall,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
            )
        }
        Text(
            text = item.displayTitle?.breakable() ?: "---",
            style = MaterialTheme.typography.bodyMedium,
            fontWeight = FontWeight.SemiBold,
        )

        val subLabel =
            buildString {
                item.quality?.let { quality ->
                    append(quality.qualityLabel)
                    bullet()
                }
                if (item !is ListenarrHistoryItem) {
                    append(item.languages.singleLanguageLabel())
                }
                item.indexerLabel?.let { indexerLabel ->
                    bullet()
                    append(indexerLabel)
                }
                if (item is ListenarrHistoryItem) {
                    append(item.message)
                }
            }
        subLabel.takeUnless { it.isEmpty() }?.let { subLabel ->
            Text(
                text = subLabel,
                style = MaterialTheme.typography.bodySmall,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
            )
        }
    }
}

File: iosApp/iosApp/Views/Components/HistoryItemView.swift
//
//  HistoryItemView.swift
//  iosApp
//
//  Created by Owen LeJeune on 2026-01-23.
//

import Shared
import SwiftUI

struct HistoryItemView: View {
    let item: HistoryItem
    
    var body: some View {
        VStack(alignment: .leading, spacing: 4) {
            HStack(spacing: 12) {
                Text(item.eventType.resource.localized())
                    .font(.system(size: 14))
                    .foregroundColor(.themePrimary)
                Text(item.date.format(pattern: "MMM d, yyyy"))
                    .font(.system(size: 12))
            }
            Text(item.displayTitle?.breakable() ?? "---")
                .fontWeight(.semibold)
            
            Text(subLabel)
                .font(.system(size: 12))
                .foregroundColor(.secondary)
        }
        .padding(.vertical, 12)
        .padding(.horizontal, 18)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(
            RoundedRectangle(cornerRadius: 10, style: .continuous)
                .fill(Color(.systemGroupedBackground))
        )
    }
    
    private var subLabel: String {
        var components: [String] = []
        
        if let qualityLabel = item.quality?.qualityLabel {
            components.append(qualityLabel)
        }
        components.append(LabelUtilsKt.singleLanguageLabel(item.languages))
        
        if let indexer = item.indexerLabel {
            components.append(indexer)
        }
        
        return components.joined(separator: " • ")
    }
}
